keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -104,7 +108,54 @@ class ApplicationService(
     }
   }
 
-  private fun getFirstDeliveryConfigFor(application: String): DeliveryConfig? =
+  /**
+   * Takes an artifact version, plus information about the type of artifact, and constructs a summary view.
+   * This should be supplemented/re-written to use actual data from stash/git/etc instead of parsing everything
+   * from the version string.
+   */
+  private fun versionToSummary(
+    artifact: DeliveryArtifact,
+    version: String,
+    environments: Set<ArtifactSummaryInEnvironment>
+  ): ArtifactVersionSummary =
+    when (artifact.type) {
+      deb -> {
+        val appversion = AppVersion.parseName(version)
+        ArtifactVersionSummary(
+          version = version,
+          environments = environments,
+          displayName = appversion?.version ?: version.removePrefix("${artifact.name}-"),
+          build = if (appversion != null) BuildMetadata(id = appversion.buildNumber.toInt()) else null,
+          git = if (appversion != null) GitMetadata(commit = appversion.commit) else null
+        )
+      }
+      docker -> {
+        var build: BuildMetadata? = null
+        var git: GitMetadata? = null
+        val dockerArtifact = artifact as DockerArtifact
+        if (dockerArtifact.hasBuild()) {
+          // todo eb: this could be less brittle
+          val regex = Regex("""^.*-h(\d+).*$""")
```

Comment:
Same comment as above on using the `TagVersionStrategy`.

Reply:
oh, answered above :D

Reply:
OK, my only other suggestion would be to move the regex to a constant/val in the companion object. 

Oh, also, if I recall correctly, the `-h{number}` part of the expression is a Netflix-specific convention where the number is the Jenkins build, no? Could we maybe change that slightly to just look for the first set of one or more consecutive digits anywhere?

```suggestion
          val regex = Regex("""^.*(\d+).*$""")
```

Reply:
I actually _only_ want this to be used here - I don't want it widely used because it's not a great way of doing things and I don't want it to spread.

Reply:
I couldn't help but read that and think of 🦠😷...
Going nuts... 😄